cloud_utils/cache/utils.py

```diff
@@ -20,35 +23,45 @@ class VersionNotFound(Exception):
 
 
 @gamla.curry
-def _write_to_versions_file(identifier: Text, hash_to_load: Text, versions_file):
+def _write_to_cache_file(
+    identifier: str,
+    hash_to_load: str,
+    filename: str,
+    lineno: int,
+    cache_file,
+):
     new_versions_dict = gamla.pipe(
-        json.load(versions_file),
+        json.load(cache_file),
         gamla.add_key_value(
             identifier,
             {
                 _HASH_VERSION_KEY: hash_to_load,
                 _LAST_RUN_TIMESTAMP: datetime.datetime.now().isoformat(),
+                "filename": filename,
```

Comment:
`**extra_fields`, and get it from outside (to remove the duplication between the signature and this `dict`)